base/mpfr.jl

```diff
@@ -119,6 +119,9 @@ end
 convert(::Type{Rational}, x::BigFloat) = convert(Rational{BigInt}, x)
 convert(::Type{AbstractFloat}, x::BigInt) = BigFloat(x)
 
+float(::Type{BigInt})   = BigFloat
+float(::Type{BigFloat}) = BigFloat
```

Comment:
I noticed that `float(Float64)` was already fast enough so didn't think about , but you're right that such method would be more general.  I don't think such method exist, I couldn't see it in `methods(float)` and
```julia
julia> @which float(Float64)
float(::Type{T}) where T<:Number in Base at float.jl:274
```
I'll update the PR, thanks.